loader/src/reference_binder.py

```diff
@@ -223,3 +168,44 @@ def load_cached_references(self, target_ref: str) -> DefaultDict[tuple, list]:
             (source_ref, ref) = json.loads(element)
             pending_refs[tuple(source_ref)].append(ref)
         return pending_refs
+
+    @staticmethod
+    def identifier_to_key(resource_type, identifier):
+        value = identifier.get("value")
+        system = identifier.get("system")
+        return f"{resource_type}:{value}:{system}"
```

Comment:
Is there no way for this to break? For instance if one doc has value "abc:def" and system "ghi" and another doc has value "abc" and system "def:ghi". I must admit this is edgy but still...
What was the inconvenient of having a serialized tuple as a key?

Reply:
The inconvenience was useless json serialization and I wanted to follow Redis convention on key formatting : https://redis.io/topics/data-types-intro.